### PR TITLE
Add table alicloud_cms_monitor_host. Closes #169

### LINKED ISSUE #169: Add table alicloud_cms_monitor_host
**References**
https://api.alibabacloud.com/?spm=5176.14113079.top-nav.19.c4fa7815F269Iq#/?product=Cms&version=2019-01-01&api=DescribeMonitoringAgentStatuses&tab=DEMO&lang=JAVA

Test need to be done through CLI - aliyun cms --help
API - list -
DescribeMonitoringAgentHosts - will provide the instance -id 
DescribeMonitoringAgentStatuses - will provide the agent status for a particular instance-id

### PR DESCRIPTION
**Pull request by rajeshbal65**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/alicloud_cms_monitor_host []

PRETEST: tests/alicloud_cms_monitor_host

TEST: tests/alicloud_cms_monitor_host
Running terraform
alicloud_vpc.named_test_resource: Refreshing state... [id=vpc-0xi0beltzt2md5otrmcn6]
alicloud_vswitch.named_test_resource: Refreshing state... [id=vsw-0xirqrcxmkaw7yshuehwc]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Refreshing state... [id=c95f7049ef4654a48825e3764639a0628]
alicloud_vpc.named_test_resource: Modifying... [id=vpc-0xi0beltzt2md5otrmcn6]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Creating...
alicloud_vpc.named_test_resource: Modifications complete after 1s [id=vpc-0xi0beltzt2md5otrmcn6]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [1m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [2m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [3m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [4m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [5m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [6m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [7m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m10s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m20s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m30s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m40s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [8m50s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Still creating... [9m0s elapsed]
alicloud_cs_managed_kubernetes.named_test_resource[0]: Creation complete after 9m1s [id=c3f091a40d1654ef9bb4faca80a25e46a]

Warning: Deprecated Resource

The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals


Apply complete! Resources: 1 added, 1 changed, 0 destroyed.

Outputs:

cluster_id = "c3f091a40d1654ef9bb4faca80a25e46a"
host_name = "worker-k8s-for-cs-c3f091a40d1654ef9bb4faca80a25e46a"
instance_id = "i-0xic0fc0m1zgc4gqtuhu"
ip_address = "10.1.1.211"
region = "us-east-1"
resource_aka = "arn:acs:cms:us-east-1:5982111499156037:host/worker-k8s-for-cs-c3f091a40d1654ef9bb4faca80a25e46a"
resource_name = "turbottest43523"

Running SQL query: test-get-query.sql
2021-06-15T19:02:09.479+0530 [TRACE] steampipe: db.EnsureDbAndStartService start
2021-06-15T19:02:09.479+0530 [TRACE] steampipe: ensure installed
2021-06-15T19:02:09.480+0530 [TRACE] steampipe: ensured installed
2021-06-15T19:02:09.480+0530 [TRACE] steampipe: GetRunStatus - loadRunningInstanceInfo returned nil
2021-06-15T19:02:09.480+0530 [TRACE] steampipe: start service
2021-06-15T19:02:10.368+0530 [TRACE] steampipe: createDbClient
2021-06-15T19:02:10.368+0530 [TRACE] steampipe: status:  &{32598 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T19:02:10.368+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=steampipe dbname=steampipe sslmode=disable
2021-06-15T19:02:10.855+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/aws@latest
2021-06-15T19:02:11.052+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/azure@latest
2021-06-15T19:02:11.125+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/gcp@latest
2021-06-15T19:02:11.202+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/ipstack@latest
2021-06-15T19:02:11.233+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/oci@latest
2021-06-15T19:02:11.292+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/slack@latest
2021-06-15T19:02:11.328+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/alicloud@latest
2021-06-15T19:02:11.473+0530 [TRACE] steampipe: RefreshConnections, updates: &{Update:map[] Delete:map[] MissingPlugins:[] RequiredConnections:map[alicloud:0xc00043e460 aws:0xc00043e280 azure:0xc00043e2d0 gcp:0xc00043e320 ipstack:0xc00043e370 oci:0xc00043e3c0 slack:0xc00043e410]}
2021-06-15T19:02:11.473+0530 [TRACE] steampipe: no connections to update
2021-06-15T19:02:11.473+0530 [TRACE] steampipe: createDbClient
2021-06-15T19:02:11.473+0530 [TRACE] steampipe: status:  &{32598 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T19:02:11.473+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=root dbname=steampipe sslmode=disable
2021-06-15T19:02:11.490+0530 [TRACE] steampipe: createDbClient
2021-06-15T19:02:11.490+0530 [TRACE] steampipe: status:  &{32598 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T19:02:11.490+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=steampipe dbname=steampipe sslmode=disable
2021-06-15T19:02:16.084+0530 [TRACE] steampipe: shutdown
2021-06-15T19:02:16.085+0530 [TRACE] steampipe: StopDB false
2021-06-15T19:02:16.085+0530 [TRACE] steampipe: signal sent terminated
[
  {
    "host_name": "worker-k8s-for-cs-c3f091a40d1654ef9bb4faca80a25e46a",
    "instance_id": "i-0xic0fc0m1zgc4gqtuhu",
    "region": "us-east-1",
    "title": "worker-k8s-for-cs-c3f091a40d1654ef9bb4faca80a25e46a"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
2021-06-15T19:02:16.172+0530 [TRACE] steampipe: db.EnsureDbAndStartService start
2021-06-15T19:02:16.172+0530 [TRACE] steampipe: ensure installed
2021-06-15T19:02:16.172+0530 [TRACE] steampipe: ensured installed
2021-06-15T19:02:16.172+0530 [TRACE] steampipe: GetRunStatus - loadRunningInstanceInfo returned nil
2021-06-15T19:02:16.172+0530 [TRACE] steampipe: start service
2021-06-15T19:02:16.784+0530 [TRACE] steampipe: createDbClient
2021-06-15T19:02:16.784+0530 [TRACE] steampipe: status:  &{32627 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T19:02:16.784+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=steampipe dbname=steampipe sslmode=disable
2021-06-15T19:02:17.273+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/oci@latest
2021-06-15T19:02:17.323+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/slack@latest
2021-06-15T19:02:17.352+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/alicloud@latest
2021-06-15T19:02:17.483+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/aws@latest
2021-06-15T19:02:17.665+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/azure@latest
2021-06-15T19:02:17.729+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/gcp@latest
2021-06-15T19:02:17.798+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/ipstack@latest
2021-06-15T19:02:17.823+0530 [TRACE] steampipe: RefreshConnections, updates: &{Update:map[] Delete:map[] MissingPlugins:[] RequiredConnections:map[alicloud:0xc0000be640 aws:0xc0000be690 azure:0xc0000be6e0 gcp:0xc0000be730 ipstack:0xc0000be780 oci:0xc0000be2d0 slack:0xc0000be5f0]}
2021-06-15T19:02:17.823+0530 [TRACE] steampipe: no connections to update
2021-06-15T19:02:17.823+0530 [TRACE] steampipe: createDbClient
2021-06-15T19:02:17.823+0530 [TRACE] steampipe: status:  &{32627 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T19:02:17.823+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=root dbname=steampipe sslmode=disable
2021-06-15T19:02:17.829+0530 [TRACE] steampipe: createDbClient
2021-06-15T19:02:17.829+0530 [TRACE] steampipe: status:  &{32627 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T19:02:17.829+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=steampipe dbname=steampipe sslmode=disable
2021-06-15T19:02:25.112+0530 [TRACE] steampipe: shutdown
2021-06-15T19:02:25.112+0530 [TRACE] steampipe: StopDB false
2021-06-15T19:02:25.112+0530 [TRACE] steampipe: signal sent terminated
[
  {
    "host_name": "worker-k8s-for-cs-c3f091a40d1654ef9bb4faca80a25e46a",
    "instance_id": "i-0xic0fc0m1zgc4gqtuhu",
    "region": "us-east-1",
    "title": "worker-k8s-for-cs-c3f091a40d1654ef9bb4faca80a25e46a"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
2021-06-15T19:02:25.196+0530 [TRACE] steampipe: db.EnsureDbAndStartService start
2021-06-15T19:02:25.196+0530 [TRACE] steampipe: ensure installed
2021-06-15T19:02:25.197+0530 [TRACE] steampipe: ensured installed
2021-06-15T19:02:25.197+0530 [TRACE] steampipe: GetRunStatus - loadRunningInstanceInfo returned nil
2021-06-15T19:02:25.197+0530 [TRACE] steampipe: start service
2021-06-15T19:02:25.828+0530 [TRACE] steampipe: createDbClient
2021-06-15T19:02:25.828+0530 [TRACE] steampipe: status:  &{32655 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T19:02:25.828+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=steampipe dbname=steampipe sslmode=disable
2021-06-15T19:02:26.321+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/oci@latest
2021-06-15T19:02:26.371+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/slack@latest
2021-06-15T19:02:26.399+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/alicloud@latest
2021-06-15T19:02:26.528+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/aws@latest
2021-06-15T19:02:26.707+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/azure@latest
2021-06-15T19:02:26.770+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/gcp@latest
2021-06-15T19:02:26.838+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/ipstack@latest
2021-06-15T19:02:26.864+0530 [TRACE] steampipe: RefreshConnections, updates: &{Update:map[] Delete:map[] MissingPlugins:[] RequiredConnections:map[alicloud:0xc0003e0820 aws:0xc0003e0870 azure:0xc0003e08c0 gcp:0xc0003e0910 ipstack:0xc0003e0960 oci:0xc0003e0730 slack:0xc0003e0780]}
2021-06-15T19:02:26.864+0530 [TRACE] steampipe: no connections to update
2021-06-15T19:02:26.864+0530 [TRACE] steampipe: createDbClient
2021-06-15T19:02:26.864+0530 [TRACE] steampipe: status:  &{32655 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T19:02:26.864+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=root dbname=steampipe sslmode=disable
2021-06-15T19:02:26.872+0530 [TRACE] steampipe: createDbClient
2021-06-15T19:02:26.872+0530 [TRACE] steampipe: status:  &{32655 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T19:02:26.872+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=steampipe dbname=steampipe sslmode=disable
2021-06-15T19:02:30.355+0530 [TRACE] steampipe: shutdown
2021-06-15T19:02:30.355+0530 [TRACE] steampipe: StopDB false
2021-06-15T19:02:30.355+0530 [TRACE] steampipe: signal sent terminated
null
✔ PASSED

Running SQL query: test-turbot-query.sql
2021-06-15T19:02:30.434+0530 [TRACE] steampipe: db.EnsureDbAndStartService start
2021-06-15T19:02:30.434+0530 [TRACE] steampipe: ensure installed
2021-06-15T19:02:30.434+0530 [TRACE] steampipe: ensured installed
2021-06-15T19:02:30.434+0530 [TRACE] steampipe: GetRunStatus - loadRunningInstanceInfo returned nil
2021-06-15T19:02:30.434+0530 [TRACE] steampipe: start service
2021-06-15T19:02:31.052+0530 [TRACE] steampipe: createDbClient
2021-06-15T19:02:31.052+0530 [TRACE] steampipe: status:  &{32683 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T19:02:31.052+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=steampipe dbname=steampipe sslmode=disable
2021-06-15T19:02:31.557+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/alicloud@latest
2021-06-15T19:02:31.687+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/aws@latest
2021-06-15T19:02:31.869+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/azure@latest
2021-06-15T19:02:31.932+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/gcp@latest
2021-06-15T19:02:32.001+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/ipstack@latest
2021-06-15T19:02:32.027+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/oci@latest
2021-06-15T19:02:32.077+0530 [DEBUG] steampipe: GetPluginPath for remote schema hub.steampipe.io/plugins/turbot/slack@latest
2021-06-15T19:02:32.105+0530 [TRACE] steampipe: RefreshConnections, updates: &{Update:map[] Delete:map[] MissingPlugins:[] RequiredConnections:map[alicloud:0xc000030370 aws:0xc000030690 azure:0xc0000306e0 gcp:0xc000030730 ipstack:0xc000030780 oci:0xc0000307d0 slack:0xc000030820]}
2021-06-15T19:02:32.105+0530 [TRACE] steampipe: no connections to update
2021-06-15T19:02:32.105+0530 [TRACE] steampipe: createDbClient
2021-06-15T19:02:32.106+0530 [TRACE] steampipe: status:  &{32683 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T19:02:32.106+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=root dbname=steampipe sslmode=disable
2021-06-15T19:02:32.112+0530 [TRACE] steampipe: createDbClient
2021-06-15T19:02:32.112+0530 [TRACE] steampipe: status:  &{32683 9193 [localhost 127.0.0.1] local query 115a-4ba8-b16f steampipe steampipe}
2021-06-15T19:02:32.112+0530 [TRACE] steampipe: Connection string:  host=localhost port=9193 user=steampipe dbname=steampipe sslmode=disable
2021-06-15T19:02:37.606+0530 [TRACE] steampipe: shutdown
2021-06-15T19:02:37.606+0530 [TRACE] steampipe: StopDB false
2021-06-15T19:02:37.606+0530 [TRACE] steampipe: signal sent terminated
[
  {
    "akas": [
      "arn:acs:cms:us-east-1:5982111499156037:host/worker-k8s-for-cs-c3f091a40d1654ef9bb4faca80a25e46a"
    ],
    "host_name": "worker-k8s-for-cs-c3f091a40d1654ef9bb4faca80a25e46a",
    "title": "worker-k8s-for-cs-c3f091a40d1654ef9bb4faca80a25e46a"
  }
]
✔ PASSED

POSTTEST: tests/alicloud_cms_monitor_host

TEARDOWN: tests/alicloud_cms_monitor_host

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### Basic info

```sql
select
  host_name,
  instance_id,
  is_aliyun_host ali_uid,
  ip_group,
  operating_system,
  region
from
  alicloud_cms_monitor_host;

+-----------------------------------------------------+------------------------+---------+-----------------------------+------------------+------------+
| host_name                                           | instance_id            | ali_uid | ip_group                    | operating_system | region     |
+-----------------------------------------------------+------------------------+---------+-----------------------------+------------------+------------+
| worker-k8s-for-cs-ce3a26021cf2747cfa1387b6410ed6764 | i-a2d0y16ulm5ti77w86el | true    | 172.18.243.7                | Linux            | ap-south-1 |
| worker-k8s-for-cs-ce3a26021cf2747cfa1387b6410ed6764 | i-a2d0y16ulm5ti77w86em | true    | 172.18.243.9                | Linux            | ap-south-1 |
| launch-advisor-20210614                             | i-a2d0y16ulm5th9odb8s4 | true    | 147.139.43.194,172.18.243.1 | Linux            | ap-south-1 |
| worker-k8s-for-cs-ce3a26021cf2747cfa1387b6410ed6764 | i-a2d0y16ulm5ti77w86ek | true    | 172.18.243.8                | Linux            | ap-south-1 |
+-----------------------------------------------------+------------------------+---------+-----------------------------+------------------+------------+
```

### List cloud monitor agent status

```sql
select
  host_name,
  m ->> 'InstanceId' as instance_id,
  m -> 'AutoInstall' as auto_install,
  m -> 'Status' as status
from
  alicloud_cms_monitor_host,
  jsonb_array_elements(monitoring_agent_status) as m;

+-----------------------------------------------------+------------------------+--------------+-----------------+
| host_name                                           | instance_id            | auto_install | status          |
+-----------------------------------------------------+------------------------+--------------+-----------------+
| worker-k8s-for-cs-ce3a26021cf2747cfa1387b6410ed6764 | i-a2d0y16ulm5ti77w86ek | true         | "not_installed" |
| launch-advisor-20210614                             | i-a2d0y16ulm5th9odb8s4 | true         | "not_installed" |
| worker-k8s-for-cs-ce3a26021cf2747cfa1387b6410ed6764 | i-a2d0y16ulm5ti77w86em | true         | "not_installed" |
| worker-k8s-for-cs-ce3a26021cf2747cfa1387b6410ed6764 | i-a2d0y16ulm5ti77w86el | true         | "not_installed" |
+-----------------------------------------------------+------------------------+--------------+-----------------+
```

### List hosts provided by Alibaba Cloud.

```sql
select
  host_name,
  is_aliyun_host
from
  alicloud_cms_monitor_host
where
  is_aliyun_host;

+-----------------------------------------------------+----------------+
| host_name                                           | is_aliyun_host |
+-----------------------------------------------------+----------------+
| worker-k8s-for-cs-ce3a26021cf2747cfa1387b6410ed6764 | true           |
| worker-k8s-for-cs-ce3a26021cf2747cfa1387b6410ed6764 | true           |
| worker-k8s-for-cs-ce3a26021cf2747cfa1387b6410ed6764 | true           |
| launch-advisor-20210614                             | true           |
+-----------------------------------------------------+----------------+
```
```
</details>

### CODE REVIEW

**Comment by rajeshbal65** [comment]
Terraform does not supports and also there is no CLI call to create a Monitoring Host. So no Integration test added.

File: alicloud-test/tests/alicloud_cms_monitor_host/test-get-expected.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[
+	{
```
Comment (bigdatasourav) [inline alicloud-test/tests/alicloud_cms_monitor_host/test-get-expected.json:2]: It should be 2 spaces.

File: alicloud-test/tests/alicloud_cms_monitor_host/test-turbot-expected.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+[
+	{
+		"akas": [
```
Comment (bigdatasourav) [inline alicloud-test/tests/alicloud_cms_monitor_host/test-turbot-expected.json:3]: It should be 2 spaces.

File: alicloud-test/tests/alicloud_cms_monitor_host/test-get-expected.json (hunk 1 above)
Comment (bigdatasourav) [reply alicloud-test/tests/alicloud_cms_monitor_host/test-get-expected.json:2]: format is not correct.

File: alicloud-test/tests/alicloud_cms_monitor_host/variables.tf [hunk 3]
```diff
@@ -0,0 +1,78 @@
+
```
Comment (bigdatasourav) [inline alicloud-test/tests/alicloud_cms_monitor_host/variables.tf:1]:
```suggestion
```

File: alicloud-test/tests/alicloud_cms_monitor_host/test-get-query.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+select host_name, instance_id, title, region
+from alicloud_cms_monitor_host
+where host_name = '{{ output.host_name.value }}' and instance_id = '{{ output.instance_id.value }}'
```
Comment (bigdatasourav) [inline alicloud-test/tests/alicloud_cms_monitor_host/test-get-query.sql:3]: key column in table is host_name then why we are checking for instance_id as well.

File: alicloud/table_alicloud_cms_monitor_host.go [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+	request.Scheme = "https"
+	request.HostName = hostName
+
+	response, err := client.DescribeMonitoringAgentHosts(request)
+	if err != nil {
+		plugin.Logger(ctx).Error("getCmsMonitorHost", "query_error", err, "request", request)
+		return nil, err
+	}
+
+	if response.Hosts.Host != nil && len(response.Hosts.Host) > 0 {
```
Comment (bigdatasourav) [inline alicloud/table_alicloud_cms_monitor_host.go:195]: do we need this check?

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**